lib/app_profiler/middleware.rb

```diff
@@ -24,13 +24,26 @@ def call(env, params = AppProfiler::RequestParameters.new(Rack::Request.new(env)
 
     def profile(env, params)
       response = nil
+      orig_backend = nil
 
       return yield unless params.valid?
 
       params_hash = params.to_h
 
       return yield unless before_profile(env, params_hash)
 
+      if params.backend
+        orig_backend = AppProfiler.profiler_backend
+        if defined?(AppProfiler::VernierBackend) &&
+            params.backend == AppProfiler::VernierBackend::NAME
+          return yield unless (AppProfiler.backend = AppProfiler::VernierBackend)
```

Comment:
`return yield unless (AppProfiler.backend = AppProfiler::VernierBackend)` I am not sure I understand why do we need this 🤔

Reply:
is it because `AppProfiler#backend=` can raise if `running?` but that is not rescued here.

I am starting to think if we are better of by doing this `running?` check in middleware and one place. Set a var to true before running the profiling code and reset before sending the response. For a multithreaded app, if running is true for another request we just do not execute the profiling code.

This assumes that no code will interact with backends directly, which I think they should not.

Reply:
yeah i think actually it would be better to just return "false" from `backend=` I realize now that raising without an ensure block is bad. The intention was that if we were not able to change the backend, but that was requested, to just run the code and fail to profile.

I agree that it's getting a bit hairy with the `running?` checks, because that is usually where the edge cases tend to be.